File: CS/WindowsApplication1/Form1.cs
using DevExpress.Utils.Drawing;
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Data;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        public void InitData()
        {
            var dt = new DataTable();
            dt.Columns.Add("BoolProperty", typeof(bool));
            dt.Columns.Add("StringProperty", typeof(string));
            dt.Columns.Add("CurrentDate", typeof(DateTime));
            dt.Columns.Add("IntProperty", typeof(Int32));
            for (int i = 0; i <= 1000; i++)
                dt.Rows.Add(new object[] { 0, i, DateTime.Today, i });
            gridControl1.DataSource = dt;
        }
        private void Form1_Load(object sender, EventArgs e)
        {
            InitData();
        }
        private void gridView1_CustomDrawColumnHeader(object sender, ColumnHeaderCustomDrawEventArgs e)
        {
            GridView view = sender as GridView;
            if (e.Column?.FieldName == view.FocusedColumn.FieldName)
                e.Info.State = ObjectState.Hot;
        }
        private void gridView1_FocusedColumnChanged(object sender, DevExpress.XtraGrid.Views.Base.FocusedColumnChangedEventArgs e)
        {
            ((GridView)sender).InvalidateColumnHeader(e.FocusedColumn);
            ((GridView)sender).InvalidateColumnHeader(e.PrevFocusedColumn);
        }
        private void gridView1_Layout(object sender, EventArgs e)
        {
            ((GridView)sender).InvalidateColumnHeader(null);
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace WindowsApplication1

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Public Sub InitData()
            Dim dt = New DataTable()
            dt.Columns.Add("BoolProperty", GetType(Boolean))
            dt.Columns.Add("StringProperty", GetType(String))
            dt.Columns.Add("CurrentDate", GetType(Date))
            dt.Columns.Add("IntProperty", GetType(Integer))
            For i As Integer = 0 To 1000
                dt.Rows.Add(New Object() {0, i, Date.Today, i})
            Next

            gridControl1.DataSource = dt
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            InitData()
        End Sub

        Private Sub gridView1_CustomDrawColumnHeader(ByVal sender As Object, ByVal e As ColumnHeaderCustomDrawEventArgs)
            Dim view As GridView = TryCast(sender, GridView)
            If Equals(e.Column?.FieldName, view.FocusedColumn.FieldName) Then e.Info.State = ObjectState.Hot
        End Sub

        Private Sub gridView1_FocusedColumnChanged(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.FocusedColumnChangedEventArgs)
            CType(sender, GridView).InvalidateColumnHeader(e.FocusedColumn)
            CType(sender, GridView).InvalidateColumnHeader(e.PrevFocusedColumn)
        End Sub

        Private Sub gridView1_Layout(ByVal sender As Object, ByVal e As EventArgs)
            CType(sender, GridView).InvalidateColumnHeader(Nothing)
        End Sub
    End Class
End Namespace
